src/Agent.Listener/Configuration/InteractiveSessionConfigurationManager.cs

```diff
@@ -0,0 +1,213 @@
+            regHelper.SetRegistry(WellKnownRegistries.StartupProcess, startupCommand);
+
+            regHelper.SetRegistry(WellKnownRegistries.ShutdownReason, "0");
+            regHelper.SetRegistry(WellKnownRegistries.ShutdownReasonUI, "0");
+
+            regHelper.SetRegistry(WellKnownRegistries.LegalNoticeCaption, "");
+            regHelper.SetRegistry(WellKnownRegistries.LegalNoticeText, "");
+
+            var processInvoker = HostContext.GetService<IProcessInvoker>();
+            processInvoker.ExecuteAsync(
+                            workingDirectory: string.Empty,
+                            fileName: "powercfg.exe",
+                            arguments: "/Change monitor-timeout-ac 0",
+                            environment: null,
+                            cancellationToken: CancellationToken.None).Wait();
+
+            processInvoker.ExecuteAsync(
+                            workingDirectory: string.Empty,
+                            fileName: "powercfg.exe",
+                            arguments: "/Change monitor-timeout-dc 0",
+                            environment: null,
+                            cancellationToken: CancellationToken.None).Wait();
+        }
+
+        public bool RestartNeeded()
+        {
+            var regHelper = new WindowsRegistryHelper(HostContext.GetService<IWindowsRegistryManager>());
+            var userName = regHelper.GetRegistry(WellKnownRegistries.AutoLogonUserName);
+            var domainName = regHelper.GetRegistry(WellKnownRegistries.AutoLogonDomainName);
+
+            if(string.IsNullOrEmpty(userName) || string.IsNullOrEmpty(domainName))
+            {
+                return false;
+            }
+            var nativeWindowsHelper = HostContext.GetService<INativeWindowsServiceHelper>();
+            return !nativeWindowsHelper.IsTheSameUserLoggedIn(domainName, userName);
+        }
+
+        public void UnConfigure()
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ShowAutoLogonWarningIfAlreadyEnabled(WindowsRegistryHelper regHelper, string userName)
+        {
+            var regValue = regHelper.GetRegistry(WellKnownRegistries.AutoLogon);
+            int.TryParse(regValue, out int autoLogonEnabled);
+            if(autoLogonEnabled == 1)
+            {
+                var autoLogonUser = regHelper.GetRegistry(WellKnownRegistries.AutoLogonUserName);
+                if(!userName.Equals(autoLogonUser, StringComparison.CurrentCultureIgnoreCase))
+                {
+                    _terminal.WriteLine(string.Format(StringUtil.Loc("AutoLogonAlreadyEnabledWarning"), userName));
+                }
+            }
+        }
+
+        //todo: move it to a utility class so that at other places it can be re-used
+        private void GetAccountSegments(string account, out string domain, out string user)
+        {
+            string[] segments = account.Split('\\');
+            domain = string.Empty;
```

Comment:
shouldn't domain default to `.\`? what happens to auto logon if user provided `testUser` as input without domain